src/main/java/io/zipcoder/Classroom.java

```diff
@@ -1,4 +1,143 @@
 package io.zipcoder;
 
+import java.util.Arrays;
+import java.util.HashMap;
+
 public class Classroom {
+
+    protected Student[] students;
+    HashMap<Student, String> gradeBook= new HashMap<>();
+
+    public Classroom() {
+
+        this.students = new Student[30];
+    }
+
+    public Classroom(int maxNumberOfStudents) {
+        this.students = new Student[maxNumberOfStudents];
+    }
+
+    public Classroom(Student[] aStudentArray) {
+        this.students = aStudentArray;
+    }
+
+    public Student[] getStudents() {
+        return this.students;
+
+    }
+
+    public double getClassTotalExamScore() {
+        double classTotalExamScore = 0;
+        for (Student student : this.students) {
+            classTotalExamScore += student.getTotalExamScore();
+        }
+        return classTotalExamScore;
+    }
+
+    public int getClassTotalCountOfExams() {
+        int classTotalCountOfExams = 0;
+        for (Student student : this.students) {
+            classTotalCountOfExams += student.getCountOfExams();
+        }
+        return classTotalCountOfExams;
+    }
+
+    public double getClassAverageExamScore() {
+        double rawAverage = getClassTotalExamScore() / getClassTotalCountOfExams();
+        double roundAndMoveDecimal = Math.round(rawAverage * 100);
+        double classAverageExamScore = roundAndMoveDecimal / 100;
+        return classAverageExamScore;
```

Comment:
once you know this works, you should collapse the steps into cleaner code.

Reply:
I originally had line 47 and 48 as one line, but it was not working properly.  I had to break it out like this to get the return I needed.